### DISCUSSION TITLE
Possibility to define the addons folder per project in the project settings, maybe more than one as well

### DISCUSSION THREAD

**Discussion by TheRealSlander** [godotengine/godot-proposals#7382] [Editor]
In a Godot project, we can personalize almost everything, but the addons folder is forced to be called "addons" (and I don't mention the fact that it does not match the "plugin" tab in the editor ^^).

This is annoying for some reasons (like naming conventions in a team for example) and it could be made a little more juicy to have the possibility to define 1 or more addons folders via a setting in the project settings page.

A useful case would be for example the use of 10 addons/plugins in a Godot project, 5 are only used in the editor to help developing and should not be exported with the game and 5 others are meant to be exported with the game. We could then define a "editor_plugins" folder and a "game_plugins" one, then put respective addons/plugins in them. In this case, it could be simple to define in the export options which folder should be exported, and which should not instead of having to set 5 separate folders.

Just a thought...

**Comment by Calinou** [comment]
Like in other programming languages, I think it's important to keep a unique convention for add-on location. It makes discovery by third-party programs *far* easier (such as asset libraries, dependency management tools, etc).

For instance, Node.js/npm uses `node_modules`, PHP's Composer uses `vendor`, etc.

If you want global add-ons, I think https://github.com/godotengine/godot-proposals/issues/831 is a better solution to go with.
